enhancements/machine-config/on-cluster-layering.md

```diff
@@ -0,0 +1,1137 @@
+  in bootc. The MCO would not be alone since RHEL For Edge and RHIVOS are likely
+  to face similar situations and friction. While these concerns exist,
+  addressing them within the context of RHEL For Edge and RHIVOS is out-of-scope
+  for this enhancement.
+
+- Giving users the ability to create custom derived OCL images, where they
+  are not required to build their image using our base image.This will
+  potentially be in scope for any future work after OCL is GA and used as the default update path.
+
+## Proposal
+
+### Workflow Description
+
+1. The cluster administrator realizes that they need to add custom content,
+   such as a software agent or device driver to the OS on each of their
+   cluster nodes.
+
+2. The cluster administrator creates a MachineOSConfig, adding a
+   Containerfile, as well as the one-time setup of what image registry to
+   push the final image to, and the credentials needed to push to this
+   registry, etc.
+
+3. The cluster administrator specifies any machine configuration changes via
+   the MachineConfig object.
+
+4. The Machine Config Controller will read the MachineConfig, check
+   reconcilability, and create a new rendered-Machineconfig.
+
+5. The BuildController detects a new build request from the new
+   MachineOSConfig/MachineConfig creation and creates a build. A new
+   MachineOSBuild object is created to track the build progress and mark the
+   targeted rendered-MachineConfig.
+
+6. The built image is pushed to the cluster administrators’ image registry of
+   choice.
+
+7. NodeController updates the `desiredImage` annotation on each node within
+   its pool according to its rules such as maximum unavailable nodes, which
+   nodes haven't been updated yet, etc.
+
+8. The MachineConfigDaemon drains the target node, applies the new OS
+   content to the node, and reboots the node upon completion.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+The primary use-case for OCL in Hypershift is for the host cluster admins to
+quickly roll out hotfixes et. al. to all of the guest cluster nodes (in
+addition to control plane nodes). The guest cluster owners do not need to
+customize their OS. Some adaptations would be required to Hypershift in
+order to traverse the host \<-\> guest cluster barrier. It is worth noting
+that within a Hypershift setup, the management cluster may be several
+versions behind the guest clusters or may not even be running OpenShift at
+all. Additionally, the guest clusters can each run different versions of
+OCP, meaning that it would not be appropriate to build a single base image
+to be used for all guest cluster nodes. Instead, images should be built for
+each guest clusters' OS / extensions image.
+
+A requirement for Hypershift support is that the API extensions described
+below should be namespaced since, to my understanding, each guest cluster
+lives within a different namespace of the management cluster.
```

Comment:
Does HyperShift have different requirements to standalone? Are there any fields on the API that would be required in standalone, but not relevant in HyperShift? Or vice versa?